File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/checkin/map/MapAction.kt
package com.hhp227.concafe.presentation.main.checkin.map

import com.hhp227.concafe.presentation.main.explore.ExploreUiState

sealed interface MapAction {
    data object ClickBack : MapAction

    data class ClickCafe(val id: String) : MapAction

    data object ClickLoginPromptSignIn : MapAction

    data object DismissLoginPrompt : MapAction

    data class UpdateRegion(val region: ExploreUiState.RegionFilter) : MapAction

    data class UpdateSearchQuery(val query: String) : MapAction
}

File: iosApp/iosApp/Presentation/Main/CheckIn/Map/MapAction.swift
//
//  MapAction.swift
//  ConCafe
//
//  Created by 홍희표 on 4/23/26.
//

import Foundation

enum MapAction {
    case backTapped
    case cafeTapped(id: String)
    case loginPromptSignInTapped
    case dismissLoginPrompt
    case regionChanged(region: ExploreUiState.RegionFilter)
    case searchQueryChanged(query: String)
}
